### ISSUE TITLE
指令精调后出现胡言乱语和好多感叹号

### ISSUE THREAD

**Issue by alanbeen** [ymcui/Chinese-LLaMA-Alpaca#499]
### 详细描述问题
![image](https://github.com/ymcui/Chinese-LLaMA-Alpaca/assets/20834883/ccb7c6a4-acca-406e-965a-fdfde4d53cdd)

训练脚本
lr=1e-4
lora_rank=8
lora_alpha=32
lora_trainable="q_proj,v_proj,k_proj,o_proj,gate_proj,down_proj,up_proj"
modules_to_save="embed_tokens,lm_head"
lora_dropout=0.05

pretrained_model="/alidata1/admin/LLaMA/modules/llama-7b-hf"
chinese_tokenizer_path="/alidata1/admin/LLaMA/modules/chinese-llama-lora-7b"
dataset_dir="/alidata1/admin/LLaMA/datasets"
per_device_train_batch_size=1
per_device_eval_batch_size=1
training_steps=100
gradient_accumulation_steps=1
output_dir="/alidata1/admin/LLaMA/Chinese-LLaMA-Alpaca/scripts/insure/sft_lora_model"
validation_file=./insure_validation/insure.json

deepspeed_config_file=ds_zero2_no_offload.json

torchrun --nnodes 1 --nproc_per_node 1 run_clm_sft_with_peft.py \
    --deepspeed ${deepspeed_config_file} \
    --model_name_or_path ${pretrained_model} \
    --tokenizer_name_or_path ${chinese_tokenizer_path} \
    --dataset_dir ${dataset_dir} \
    --validation_split_percentage 0.001 \
    --per_device_train_batch_size ${per_device_train_batch_size} \
    --per_device_eval_batch_size ${per_device_eval_batch_size} \
    --do_train \
    --do_eval \
    --seed $RANDOM \
    --fp16 \
    --max_steps ${training_steps} \
    --lr_scheduler_type cosine \
    --learning_rate ${lr} \
    --warmup_ratio 0.03 \
    --weight_decay 0 \
    --logging_strategy steps \
    --logging_steps 10 \
    --save_strategy steps \
    --save_total_limit 3 \
    --evaluation_strategy steps \
    --eval_steps 250 \
    --save_steps 500 \
    --gradient_accumulation_steps ${gradient_accumulation_steps} \
    --preprocessing_num_workers 8 \
    --max_seq_length 500 \
    --output_dir ${output_dir} \
    --overwrite_output_dir \
    --ddp_timeout 30000 \
    --logging_first_step True \
    --lora_rank ${lora_rank} \
    --lora_alpha ${lora_alpha} \
    --trainable ${lora_trainable} \
    --lora_dropout ${lora_dropout} \
    --torch_dtype float16 \
    --validation_file ${validation_file} \
    --ddp_find_unused_parameters False

数据集(数据量在500条左右)：
[
....
  {
    "instruction": "有安装起搏器能投保众安尊享e生2023版中端医疗险吗",
    "input": "",
    "output": "心脏起搏器的安装是在局部麻醉下进行的，手术全程中患者意识都是清楚的。一般选择锁骨下静脉为穿刺点，建立静脉通路，将起搏器的导线置入心腔内。 如果是双腔起搏器，则一根电极置入右心耳，另一根电极置入右心室，从X线下判断电极的植入位置。然后将起搏器的电极与起搏器相连接，最后在胸部切开皮肤，制作囊袋，将起搏器埋入其中后将皮肤缝合。\n\n被保险人既往症如有安装起搏器,无法投保众安尊享e生2023版中端医疗险。"
  }
.....
]

训练结束后
mv /alidata1/admin/LLaMA/Chinese-LLaMA-Alpaca/scripts/insure/sft_lora_model/pytorch_model.bin /alidata1/admin/LLaMA/Chinese-LLaMA-Alpaca/scripts/lora_model/adapter_model.bin
cp /alidata1/admin/LLaMA/modules/chinese-llama-plus-lora-7b/*token* lora_model/
cp /alidata1/admin/LLaMA/modules/chinese-llama-plus-lora-7b/adapter_config.json lora_model/

合并数据
python /alidata1/admin/LLaMA/Chinese-LLaMA-Alpaca/scripts/merge_llama_with_chinese_lora.py \
    --base_model /alidata1/admin/LLaMA/modules/llama-7b-hf \
    --lora_model /alidata1/admin/LLaMA/modules/chinese-llama-plus-lora-7b,/alidata1/admin/LLaMA/modules/chinese-alpaca-plus-lora-7b,/alidata1/admin/LLaMA/Chinese-LLaMA-Alpaca/scripts/lora_model \
    --output_type huggingface \
    --output_dir  /alidata1/admin/LLaMA/lora/

启动合并后的lora
python3 server.py --model lora --listen --api --chat --auto-devices 

帮忙分析下是什么原因，训练后不能正常回答

**Comment by airaria** [comment]
是从原版lllama开始训练的？没有经过中文预训练的话，在中文上自然效果不好

**Comment by alanbeen** [comment]
是基于原版的，但是我合并了中文基础模型
 /alidata1/admin/LLaMA/modules/chinese-llama-plus-lora-7b,
/alidata1/admin/LLaMA/modules/chinese-alpaca-plus-lora-7b
 我这里合并中文模型

**Comment by airaria** [comment]
你的脚本里的model_name_or_path指向的是/alidata1/admin/LLaMA/modules/llama-7b-hf，这个是原版llama吗

**Comment by alanbeen** [comment]
> 你的脚本里的model_name_or_path指向的是/alidata1/admin/LLaMA/modules/llama-7b-hf，这个是原版llama吗

这个是的 是要预训练才行吗

**Comment by airaria** [comment]
需要换成预训练后的模型，比如合并后的Chinese-LLaMA-plus

**Comment by csongqi** [comment]
> > 你的脚本里的model_name_or_path指向的是/alidata1/admin/LLaMA/modules/llama-7b-hf，这个是原版llama吗
> 
> 这个是的 是要预训练才行吗

训练效果怎么样？我现在也遇到类似问题，7b模型在3090单卡上指令精调速度非常快，参数和你设置差不多，且预训练模型使用的是Chinese-alpaca-plus。但是训练出来后，就是随便问个问题就是两三个字重复，完全没法使用。

**Comment by majjmajj** [comment]
> 



> 需要换成预训练后的模型，比如合并后的Chinese-LLaMA-plus

请问如果精调Chinese-Alpaca-Plus，是model_name_or_path指向的是合并后的Chinese-Alpaca-Plus吗？
